prediction_market_agent/agents/arbitrage_agent/deploy.py

```diff
@@ -0,0 +1,193 @@
+import typing as t
+
+from langchain_core.output_parsers import PydanticOutputParser
+from langchain_core.prompts import PromptTemplate
+from langchain_core.runnables import RunnableSerializable
+from langchain_openai import ChatOpenAI
+from prediction_market_agent_tooling.deploy.agent import (
+    MAX_AVAILABLE_MARKETS,
+    DeployableTraderAgent,
+)
+from prediction_market_agent_tooling.gtypes import Probability
+from prediction_market_agent_tooling.markets.agent_market import (
+    AgentMarket,
+    FilterBy,
+    SortBy,
+)
+from prediction_market_agent_tooling.markets.data_models import (
+    BetAmount,
+    Position,
+    ProbabilisticAnswer,
+    TokenAmount,
+    Trade,
+    TradeType,
+)
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.markets.omen.omen import OmenAgentMarket
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
+    OmenSubgraphHandler,
+)
+from prediction_market_agent_tooling.tools.langfuse_ import (
+    get_langfuse_langchain_config,
+    observe,
+)
+
+from prediction_market_agent.agents.arbitrage_agent.data_models import (
+    CorrelatedMarketPair,
+    Correlation,
+)
+from prediction_market_agent.db.pinecone_handler import PineconeHandler
+from prediction_market_agent.utils import APIKeys
+
+
+class DeployableOmenArbitrageAgent(DeployableTraderAgent):
+    """Agent that places mirror bets on Omen for (quasi) risk-neutral profit."""
+
+    correlation_threshold: float = 0.8
```

Comment:
I also like (2) better, but markets that are obviously strongly correlated, e.g. like in the test
```
 m1.question = "Will Kamala Harris win the US presidential election?"
m1.question = "Will Kamala Harris become the US president in 2025?"
```
only deliver 0.8 correlation.

Reply:
I ran it with gpt-4o instead of 4o-mini and it gives 1.0!!

```
from prediction_market_agent.agents.arbitrage_agent.deploy import (
    DeployableOmenArbitrageAgent,
)

agent = DeployableOmenArbitrageAgent()
correlation = agent.chain.invoke(
    {
        "main_market_question": "Will Kamala Harris win the US presidential election?",
        "related_market_question": "Will Kamala Harris become the US president in 2025?",
    }
)
print(correlation)
```

Maybe use that instead? Should be very cheap to run these prompts. Have you checked?

Reply:
Sounds like a good approach, implemented